dataprofiler/tests/data_readers/test_csv_data.py

```diff
@@ -175,5 +184,47 @@ def test_header_check_files(self):
         CSVData.is_match(path, options)
         self.assertEqual(0, options.get("header"))
 
+        # File with header, but the csv sniffer fails to detect
+        # set to 0 for auto determine
+        path = os.path.join(test_dir, 'csv/small-num.csv')
+        options = dict()
+        CSVData.is_match(path, options)
+        self.assertEqual(0, options.get("header"))
+
+        # File with header, but the csv sniffer fails to detect
+        # set to 0 for auto determine
+        path = os.path.join(test_dir, 'csv/names-col.txt')
+        options = dict()
+        CSVData.is_match(path, options)
+        self.assertEqual(0, options.get("header"))
+
+        # File with header, but the csv sniffer fails to detect
+        # set to 0 for auto determine
+        path = os.path.join(test_dir, 'csv/names-col-empty.txt')
+        options = dict()
+        CSVData.is_match(path, options)
+        self.assertEqual(0, options.get("header"))
+
+        # File with header, but the csv sniffer fails to detect
+        # set to 0 for auto determine
+        path = os.path.join(test_dir, 'csv/names-col-without-space.txt')
+        options = dict()
+        CSVData.is_match(path, options)
+        self.assertEqual(0, options.get("header"))
+
+        # File with header, but the csv sniffer fails to detect
+        # set to 0 for auto determine
+        path = os.path.join(test_dir, 'csv/two-header.txt')
+        options = dict()
+        CSVData.is_match(path, options)
+        self.assertEqual(1, options.get("header"))
+
+        # File with header, but the csv sniffer fails to detect
+        # set to 0 for auto determine
+        path = os.path.join(test_dir, 'csv/sparse-interchange-none.txt')
+        options = dict()
+        CSVData.is_match(path, options)
+        self.assertEqual(0, options.get("header"))
```

Comment:
Could we make this a for loop? This seems a bit repetitive. Maybe define a list of tuples like [(path, header), ...] where you store the file path and the expected value for options.get("header")?

Reply:
Or could you even just index into the list of dictionaries defined in input_file_names?

Reply:
yeah, I can make it as a for loop